fabric-common/lib/IdentityContext.js

```diff
@@ -84,11 +84,11 @@ const IdentityContext = class {
 	 * @param {Buffer} payload - The payload bytes that require a signature
 	 * @return {Buffer} - The signature in bytes
 	 */
-	sign(payload = checkParameter('payload')) {
+	async sign(payload = checkParameter('payload')) {
 		const method = 'sign';
 		logger.debug('%s - start', method);
 		const signer = this.user.getSigningIdentity();
-		const signature = Buffer.from(signer.sign(payload));
+		const signature = Buffer.from(await Promise.resolve(signer.sign(payload)));
```

Comment:
What is different between `await` and `await(Promise.resolve(..))`

Reply:
Current cryptosuite of `Default X509` and `HSM X509` return a simple `Buffer` and this PR wants to allow `sign()` method to return a `Promise` also. So I thought `promise.resolve()` will convert a `sync sign()` method of those two suite into a `async` method. But I guess `await sign()` will work for both `sync` and `async`.